insights/core/reducer.py

```diff
@@ -132,15 +132,16 @@ def _run(reducers, local_output, shared_output, error_handler, output_dict=None,
             continue
         real_module = sys.modules[func.__module__]
         local = local_output[real_module]
-        log_component_outputs(func, local, shared_output)
-        r = run_reducer(func, local, shared_output, error_handler, stats=stats)
-        if r:
-            shared_output[func] = r
-            logger.debug("Reducer output: %s", r)
-            if func.shared and output_dict:
-                output_dict[func] = r
-            if func in plugins.EMITTERS:
-                yield func, r
+        log_parser_outputs(func, local, shared_output)
+        if func not in shared_output:
```

Comment:
I'd like to see a new PR for this block of changes, just so I can understand why you'd like to port it.